base/complex.jl

```diff
@@ -1028,3 +1028,114 @@ function complex(A::AbstractArray{T}) where T
     end
     convert(AbstractArray{typeof(complex(zero(T)))}, A)
 end
+
+"""
+    div(z1, z2, r::RoundingMode=RoundNearest)
+
+The quotient from Euclidean division of two Gaussian integers. 
+Computes z1/z2, rounded to a Gaussian integer according to the rounding mode `r`. 
+In other words, the quantity
+
+    round(z1/z2,r,r)
+
+without any intermediate rounding.
+I.e. quotient with the same rounding mode `r` applied to real and imaginary parts.
+
+# Examples:
+```jldoctest
+julia> div(4 + 4im, 3 + 1im, RoundDown)
+1 + 0im
+
+julia> div(4 + 4im, 3 + 1im, RoundUp)
+2 + 1im
+
+julia> div(5 + 1im, 2 + 1im, RoundNearest)
+2 - 1im
+
+julia> div(5 + 1im, 2 + 1im, RoundNearestTiesAway)
+2 - 1im
+
+julia> div(-5 + 5im, 2 + 2im, RoundNearest)
+0 + 2im
+
+julia> div(-5 + 5im, 2 + 2im, RoundNearestTiesAway)
+0 + 3im
+
+julia> div(-5 + 5im, 2 + 2im, RoundNearestTiesUp)
+0 + 3im
+```
+"""
+function div(a::Complex{T}, b::Complex{V}, r::RoundingMode=RoundNearest) where {T<:Integer, V<:Integer}
+    R = promote_type(T, V)
+    a, b = Complex{R}(a), Complex{R}(b)
+    b̅ = conj(b)
+    # TODO: Handle overflow when calculating a*b̅
+    t = a*b̅
+    # TODO: Create checked_abs2(::Complex{<:Integer})
+    # TODO: Handle overflow when calculating the norm of complex numbers
+    abs2_b = abs2(b)
+    abs2_b < 0 && __throw_gcd_overflow(a, b)
+    Complex(div(real(t), abs2_b, r), div(imag(t), abs2_b, r))
+end
+
+"""
+    rem(z1, z2, r::RoundingMode=RoundNearest)
+
+Compute the remainder of `z1` after Euclidean division by `z2`, with the quotient 
+rounded according to the rounding mode `r`. In other words, the quantity
+
+    z1 - z2*round(z1/z2,r,r)
+
+without any intermediate rounding.
+"""
+function rem(a::Complex{T}, b::Complex{V}, r::RoundingMode=RoundNearest) where {T<:Integer, V<:Integer}
+    R = promote_type(T, V)
+    a, b = Complex{R}(a), Complex{R}(b)
+    a - b * div(a, b, r)
+end
+
+function _first_quadrant(a::Complex)
+    ar, ai = reim(a)
+    if iszero(ar)
+        Complex(abs(ai), zero(ar))
+    elseif iszero(ai)
+        Complex(abs(ar), zero(ai))
+    elseif ar > 0 && ai > 0     # The complex number is already in the first quadrant
+        a
+    elseif ar < 0 && ai > 0     # In the second quadrant
+        Complex(ai, -ar)
+    elseif ar < 0 && ai < 0     # In the third quadrant
+        -a
+    else                        # In the fourth quadrant
+        Complex(-ai, ar)
+    end
+end
+
+"""
+    gcd(z1, z2)
```

Comment:
If you implement `gcd` wouldn't you also expect `gcdx` (as for `Rational`)?

Is there a sensible extension for `Complex{<:Rational}` ?

Reply:
As it turns out, yes, gcdx can also be extended to `Complex{<:Integer}`.
https://math.stackexchange.com/questions/2911332/modular-inverse-of-gaussian-integers/2913973#2913973

Also, I think rather than `Complex{<:Rational}`, you meant `Complex{<:Integer}`, since the current methods are being added for `Complex{<:Integer}` only. In future, we might consider extending them to `Complex{<:Rational}`.

Reply:
Should I add gcdx method as well?